src/node_report.cc

```diff
@@ -1030,14 +1039,20 @@ static void PrintResourceUsage(std::ostream& out) {
 #if defined(__APPLE__) || defined(_AIX)
     snprintf( buf, sizeof(buf), "%ld.%06d", stats.ru_utime.tv_sec, stats.ru_utime.tv_usec);
     out << "\n  User mode CPU: " << buf << " secs";
+    cpu_abs = std::stod(buf);
     snprintf( buf, sizeof(buf), "%ld.%06d", stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
     out << "\n  Kernel mode CPU: " << buf << " secs";
+    cpu_abs += std::stod(buf);
 #else
     snprintf( buf, sizeof(buf), "%ld.%06ld", stats.ru_utime.tv_sec, stats.ru_utime.tv_usec);
     out << "\n  User mode CPU: " << buf << " secs";
+    cpu_abs = std::stod(buf);
     snprintf( buf, sizeof(buf), "%ld.%06ld", stats.ru_stime.tv_sec, stats.ru_stime.tv_usec);
     out << "\n  Kernel mode CPU: " << buf << " secs";
+    cpu_abs += std::stod(buf);
 #endif
+    cpu_percentage = (cpu_abs / diff_time) * 100;
+    out << "\n  Average Consumption : "<< cpu_percentage << "%";
```

Comment:
I'm probably being paranoid but could you multiply by 100.0 to make sure there are no conversions back to integer arithmetic.
Could you change the label to  "Average CPU Consumption"? When I wrote the report out this came just before the resident set size line and it wasn't clear whether it was average CPU or memory consumption. It might be worth specifying the format for the percentage more tightly so we don't get things like 1.9e-09 if there's a very small amount of CPU used.
It's also probably worth writing out diff_time as "Uptime" or similar as that's a) useful to know and b) means someone can see where the percentage came from.